src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -2071,6 +2104,45 @@ bool WindowImplX11::processEvent(XEvent& windowEvent)
 
             break;
         }
+
+        // Mouse moved raw
+        case GenericEvent:
+        {
+            if (XGetEventData(m_display, &windowEvent.xcookie)
+                    && windowEvent.xcookie.evtype == XI_RawMotion)
+            {
+                const XIRawEvent* rawEv;
+                rawEv = static_cast<const XIRawEvent*>(windowEvent.xcookie.data);
+
+                const int mouseAxis = 2;
+                int relativeValues[2] = { 0, 0 };
+                int limit = rawEv->valuators.mask_len;
+
+                if (limit > mouseAxis)
+                    limit = mouseAxis;
+                limit *= 8;
+
+                // Get relative input values
+                const double* inputValues = rawEv->raw_values;
+                for (int i = 0; i < limit; ++i)
+                {
+                    if (XIMaskIsSet(rawEv->valuators.mask, i))
+                    {
+                        relativeValues[i] = static_cast<int>(*inputValues);
+                        ++inputValues;
+                    }
+                }
```

Comment:
If `limit` happened to be larger than 2 we could end up writing past the end of the array.